modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,32 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
+				var webp_sizes = window.wpPerfLabWebpSizes;
+				var size_keys = Object.keys( webp_sizes );
 
 				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
+					console.log( media[i] );
 					var srcset = images[j].getAttribute( 'srcset' );
 					if ( srcset ) {
-						images[j].setAttribute( 'srcset', srcset.replace( /\.webp(\s)/ig, ext[0] + '$1' ) );
+						// Update full image in srcset.
+						srcset = srcset.replace( images[j].src, media[i].media_details.sizes.full.source_url );
```

Comment:
Here the we should check the `media[i].media_details.sizes.full.source_url` is set or not to avoid undefined errors. And we should get the output source url into variable and then use them later in the code.

Reply:
Thank you for the review. I have updated it. Please re-review it.